src/main/java/com/alex/eshop/keycloak/KeycloakService.java

```diff
@@ -25,4 +41,66 @@ public AccessTokenResponse getToken() {
         AuthzClient authzClient = AuthzClient.create(authzConf);
         return authzClient.obtainAccessToken();
     }
+
+    public RealmResource realmResource() {
+        return keycloakClientFactory.getInstance()
+                .realm(applicationProperties.getKeycloak().getRealm());
+    }
+
+    public UsersResource usersResource() {
+        return keycloakClientFactory.getInstance()
+                .realm(applicationProperties.getKeycloak().getRealm())
+                .users();
+    }
+
+    public List<RoleRepresentation> getRoleRepresentation() {
+        List<RoleRepresentation> realmRoles = new ArrayList<>();
+
+        realmRoles.add(realmResource().roles().get(Role.USER).toRepresentation());
+
+        return realmRoles;
+    }
+
+    public List<String> getUserRole() {
+        return getRoleRepresentation().get(0).getName().lines().toList();
+    }
+
+    public UserRepresentation getUserRepresentation(String userUuid) {
+        return usersResource()
+                .get(userUuid)
+                .toRepresentation();
+    }
+
+    public UserRepresentation createUser(UserRegisterDTO userRegisterDTO) {
+        if (!Objects.equals(userRegisterDTO.getPassword(), userRegisterDTO.getConfirmPassword())) {
+            throw new InvalidDataException("Invalid password");
```

Comment:
Exception message is uninformative. Better "Password and Confirm Password does not match"